Riot/Modules/Room/Location/RoomTimelineLocationView.swift

```diff
@@ -76,7 +108,8 @@ class RoomTimelineLocationView: UIView, NibLoadable, Themable, MGLMapViewDelegat
     
     public func displayLocation(_ location: CLLocationCoordinate2D,
```

Comment:
You can set this function private and use to separate methods. We should also handle all states of current user live location sharing. For example:
```
struct RoomTimelineLocationViewData {
	let location: CLLocationCoordinate2D
	let userAvatarData: AvatarViewData?
	let mapStyleURL: URL
}

struct LiveLocationBannerViewData {
     let icon: UIImage
     let title: String
     let timeLeftString: String?
     let rightButtonTitle: String?

     var showRightButton: Bool {
     	return rightButtonTitle != nil
     }
}

enum TimelineLiveLocationViewState {	
	case incoming(_ status: LiveLocationSharingStatus) // live location started from current user
	case outgoing(_ expiringDate: Date) // live location started by other users
}


enum LiveLocationSharingStatus {
	case starting
	case started(_ timeleft: TimeInterval)
	case failure
	case stopped
}

public func displayStaticLocation(with viewData: RoomTimelineLocationViewData) {

}

public func displayLiveLocation(with viewData: RoomTimelineLocationViewData, liveLocationViewState: TimelineLiveLocationViewState) {

let bannerViewData = self.liveLocationBannerViewData(from viewState: TimelineLiveLocationViewState)
...	
}
```